File: dotnet/Filestore/CatalogueReader.cs
namespace Filestore {

	using System.IO;
	using System.Collections.Generic;

public class CatalogueReader {

	private string filename;

	public CatalogueReader(string filename) {
		this.filename = filename;
	}

	public List<string[]> readAll() {
		List<string[]> lines = new List<string[]>();
		try {
			foreach (string line in File.ReadLines(filename)) {
				string[] args = line.Split('^');
				lines.Add(args);
			}
		} catch (IOException e) {
			throw new DataFileException(filename);
		}
		return lines;
	}

}
}

File: ruby/lib/filestore/catalogue_reader.rb
require_relative './data_file_exception'

module Filestore

	class CatalogueReader

		def initialize(filename)
			@filename = filename
		end

		def readAll
			lines = []
			begin
				File.foreach(@filename) do |line|
					lines << line.split(/\^/)
				end
			rescue Exception => e
				throw DataFileException.new(@filename, e)
			end
			lines
		end

	end
end
